services/navigation/builtin/builtin_test.go

```diff
@@ -397,3 +398,39 @@ func TestStartWaypoint(t *testing.T) {
 		})
 	})
 }
+
+func TestValidateGeometry(t *testing.T) {
+	cfg := Config{
+		BaseName:           "base",
+		MovementSensorName: "localizer",
+	}
+
+	t.Run("fail case", func(t *testing.T) {
+		cfg = CreateBox(t, cfg, r3.Vector{10, 10, 10})
+		_, err := cfg.Validate("")
+		expectedErr := "geometries specified through the navigation are not allowed to have a translation"
+		test.That(t, err.Error(), test.ShouldEqual, expectedErr)
+	})
+
+	t.Run("success case", func(t *testing.T) {
+		cfg = CreateBox(t, cfg, r3.Vector{})
+		_, err := cfg.Validate("")
+		test.That(t, err, test.ShouldBeNil)
+	})
+}
+
+func CreateBox(t *testing.T, cfg Config, translation r3.Vector) Config {
```

Comment:
This shouldn't be public.  And since this is only used in the one function you wrote can we just move it into that function?

Reply:
Also would it be possible to simplify a little?  I think we could even just manually instantiate a couple of GeoObstacleConfigs with the only difference being the translation offset of the geometry